compiler/AST/LoopExpr.cpp

```diff
@@ -169,6 +169,30 @@ GenRet LoopExpr::codegen() {
 }
 
 
+//
+// Remove the ref types and the autocopy functions for the types
+// that are about to be removed when removing 'block'.
+//
+// Todo: we probably want to do this for all types that have been
+// deleted, as a cleanup function at the end of resolve().
+//
```

Comment:
I really don't see the point in storing this in LoopExpr.cpp. I agree with the sentiment in your PR message that we should pull the dead-thing-removal code out of functionResolution.cpp into its own file in resolution/. However putting this  function in LoopExpr.cpp will only make that process harder in the future. So, please move it into functionResolution.cpp next to the others.